### PR TITLE
Restored serialization behaviour for time.Time values

### PR DESCRIPTION
**Pull request by yuce**
Previously, we used a heuristic to map `time.Time` values to the correct reference implementation type. This PR reverses that change, and uses the v1.1.1 behaviour.

### CODE REVIEW

**Review by utku-caglayan — APPROVED** [review]
LGTM!

File: client_internals_it_test.go [hunk 1]
```diff
@@ -238,14 +238,6 @@ func TestClientInternal_OrderedMembers(t *testing.T) {
 	it.Eventually(t, func() bool {
 		return sameMembers(targetUUIDs, ci.OrderedMembers())
 	})
-	types.NewUUID()
```
Comment (emreyigit) [inline client_internals_it_test.go:241]: Is this related to the PR?

Comment (yuce) [reply client_internals_it_test.go:241]: No, this is an accidental leftover from an earlier PR.

**Review by emreyigit — APPROVED** [review]
